lib/types.ts

```diff
@@ -0,0 +1,51 @@
+// Common type aliases for better code readability
+
+import type { Config as JestConfig } from '@jest/types';
+import type * as JestRunner from 'jest-runner';
+
+export type Milliseconds = number;
+export type Path = string;
+
+export interface TestDetail {
+  title: string;
+  path: Path;
+}
+
+// Copied and adapted from https://github.com/facebook/jest/blob/2dafb09d51584d3785f3280f569784ec4334b5d8/packages/jest-runner/src/index.ts#L48-L285
+export declare abstract class TestRunner {
```

Comment:
is this something we should export from Jest?

Reply:
Actually it is exported from `"jest-runner"` as
```ts
declare class JestRunner { /* ... */ }
export = JestRunner
```

But unfortunately I cannot use this `JestRunner` to annotate the return type of `createTestRunner` since it contains properties that our `BaseTestRunner` does not have, namely:
- `private readonly eventEmitter: Emittery.Typed<JestTestEvents>`
- `readonly __PRIVATE_UNSTABLE_API_supportsEventEmitters__: boolean`
- `readonly isSerial?: boolean`
- `on: Emittery.Typed<JestTestEvents>` (an alias of eventEmitter but bound to `this.eventEmitter`) 

I actually dug into the `jest` codebase but it seems that the event emitter api seems to be [considered unstable and prone to change in `jest` v27](https://github.com/facebook/jest/blob/f9132f9106e3f1b3ae6f81d1f36222055c3fe111/packages/jest-runner/src/index.ts#L102) (I suspect it has something to do with the `jest-circus` and flux-based architecture thingy), and I'm not sure if it is the right thing to add an additional dependency `emittery` to the project.

I couldn't find related docs or changelogs for the eventEmitters which were added since [this commit](https://github.com/facebook/jest/blob/eb23ac1ff0d6c5ee6b9a253724dc9f868a2133bd/packages/jest-runner/src/index.ts) so I decided I would copy and paste the `JestRunner` class declaration from `jest-runner` package with the fields that `BaseTestRunner` does not implement removed.

Reply:
I tried and I can change `TestRunner` in `lib/types.ts` to
```ts
export declare abstract class TestRunner
  implements
    Omit<JestRunner, '__PRIVATE_UNSTABLE_API_supportsEventEmitters__' | 'on'> {
  constructor(globalConfig: JestConfig.GlobalConfig);

  runTests(
    tests: Array<JestRunner.Test>,
    watcher: JestRunner.TestWatcher,
    onStart: JestRunner.OnTestStart,
    onResult: JestRunner.OnTestSuccess,
    onFailure: JestRunner.OnTestFailure,
    options: JestRunner.TestRunnerOptions,
  ): Promise<void>;
}
```

But I'm not sure if using the `Omit<>` utility type here is good practice. Any suggestions on improvements?